l3backend/l3backend-pdf.dtx

```diff
@@ -571,38 +571,35 @@
 % \end{macro}
 % \end{macro}
 %
-% \begin{macro}{\@makecol@hook}
-%   Hooks to allow link breaking: something will be needed in format mode
-%   at some stage. At present this code is disabled as there is an open
-%   question about the name of the hook: to be resolved at the \LaTeXe{}
-%   end.
+% \begin{macro}{\@makecol}
+%   Install the necessary code to break links.
 %    \begin{macrocode}
-\use_none:n
+\bool_lazy_and:nnT
+  { \cs_if_exist_p:N \hook_gput_code:nnn }
+  { \cs_if_exist_p:N \@makecol }
```

Comment:
I think, for now hooking into the generic hook is fine. Eventually this should be moved to a kernel hook once \@makecol has one (not just in latex-lab)

Reply:
I tried the following and it works fine, but imho without the pdfmanagement it is not of much use: hyperref then can't/doesn't use the backend commands and the l3pdfannot commands are not available. Wouldn't it then make more sense to tie it directly to the `\@makecol` redefinition in new-or-1 or -2?

~~~~
\DocumentMetadata{}
\documentclass{book}
\usepackage{hyperref}
\textwidth2cm
\begin{document}
xx \href{url}{long long long long long long long}

\ExplSyntaxOn
\pdfannot_link:nnn { URI }
{/A<</Type/Action/S/URI/URI(https://www.latex-project.org)>>}
{ a~very~long~link~with~lots~of~text }
\ExplSyntaxOff

\end{document}
~~~~

Reply:
> I tried the following and it works fine, but imho without the pdfmanagement it is not of much use: hyperref then can't/doesn't use the backend commands and the l3pdfannot commands are not available. Wouldn't it then make more sense to tie it directly to the `\@makecol` redefinition in new-or-1 or -2?
> 
> ```
> \DocumentMetadata{}
> \documentclass{book}
> \usepackage{hyperref}
> \textwidth2cm
> \begin{document}
> xx \href{url}{long long long long long long long}
> 
> \ExplSyntaxOn
> \pdfannot_link:nnn { URI }
> {/A<</Type/Action/S/URI/URI(https://www.latex-project.org)>>}
> { a~very~long~link~with~lots~of~text }
> \ExplSyntaxOff
> 
> \end{document}
> ```

The idea was that nothing should happen unless the user opts-in using the testphase: did I get that wrong?

Reply:
@josephwright sorry, I'm not quite sure what the question is here. But as soon as people use \DocumentMetadata they get the pdf management and hyperref uses it (and transparent and pdflscape). If you want to tie the breaking links to an opt-in, then the change to \@makecol should be made in new-or, then they get it only with a testphase key.

Reply:
@u-fischer I mean that the plan was that for `expl3`-based code, the change works 'out of the box' (as `expl3` hyperlinks are clearly 'new' material), but for `hyperref` it would only be people explicitly using the testphase, as otherwise `hyperref` doesn't use the `expl3` driver. That not correct?

Reply:
@josephwright no. People get the new hyperref driver as soon as they use the pdfmanagement. It is not possible otherwise. Packages that do PDF stuff *must* use it, or they wouldn't be compatible. This is the case since now around two years and nobody complained (some people explicitly use it only to get nicer hyperref colors ;-))

Reply:
@u-fischer  But that's _is_ testphase: until that management is built into the kernel, it remains opt-in, so most uses will not be impacted.